Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java

```diff
@@ -0,0 +1,428 @@
+                                filterSupportedPaymentMethods(configuration, isGooglePayEnabled);
+                            callback.onResult(availablePaymentMethods, null);
+                        }
+                    });
+                } else {
+                    List<DropInPaymentMethodType> availablePaymentMethods =
+                            filterSupportedPaymentMethods(configuration, false);
+                    callback.onResult(availablePaymentMethods, null);
+                }
+            }
+        });
+    }
+
+    private List<DropInPaymentMethodType> filterSupportedPaymentMethods(Configuration configuration, boolean showGooglePay) {
+        List<DropInPaymentMethodType> availablePaymentMethods = new ArrayList<>();
+
+        if (dropInRequest.isPayPalEnabled() && configuration.isPayPalEnabled()) {
+            availablePaymentMethods.add(DropInPaymentMethodType.PAYPAL);
+        }
+
+        if (dropInRequest.isVenmoEnabled() && configuration.isVenmoEnabled()) {
+            availablePaymentMethods.add(DropInPaymentMethodType.PAY_WITH_VENMO);
+        }
+
+        if (dropInRequest.isCardEnabled()) {
+            Set<String> supportedCardTypes =
+                    new HashSet<>(configuration.getSupportedCardTypes());
+            if (!configuration.isUnionPayEnabled()) {
+                supportedCardTypes.remove(DropInPaymentMethodType.UNIONPAY.getCanonicalName());
+            }
+            if (supportedCardTypes.size() > 0) {
+                availablePaymentMethods.add(DropInPaymentMethodType.UNKNOWN);
+            }
+        }
+
+        if (showGooglePay) {
+            if (dropInRequest.isGooglePaymentEnabled()) {
+                availablePaymentMethods.add(DropInPaymentMethodType.GOOGLE_PAYMENT);
+            }
+        }
+        return availablePaymentMethods;
+    }
+
+    public void launchDropInForResult(FragmentActivity activity, int requestCode) {
+        Intent intent = new Intent(activity, DropInActivity.class)
+                .putExtra(EXTRA_CHECKOUT_REQUEST, dropInRequest)
+                .putExtra(EXTRA_SESSION_ID, braintreeClient.getSessionId())
+                .putExtra(EXTRA_AUTHORIZATION, braintreeClient.getAuthorization().toString());
+        activity.startActivityForResult(intent, requestCode);
+    }
+
+    /**
+     * Called to get a user's existing payment method, if any. If your user already has an existing
+     * payment method, you may not need to show Drop-In.
+     * <p>
+     * Note: a client token must be used and will only return a payment method if it contains a
+     * customer id.
+     *
+     * @param activity the current {@link AppCompatActivity}
+     * @param callback callback for handling result
+     */
+    public void fetchMostRecentPaymentMethod(FragmentActivity activity, final FetchMostRecentPaymentMethodCallback callback) {
+        // TODO: consider sending back empty result if tokenization key auth
```

Comment:
I think throwing an error makes sense here. If merchants are calling this method with a tokenization key, they should find out right away that it won't work.

Reply:
Ok we will keep returning an error to the callback here then. Removed this TODO since it no longer applies.